index.js

```diff
@@ -0,0 +1,217 @@
+// 1
+function getDeltaFunction(deltaObject) {
+  return (subDelta = {}) => {
+    const delta = {...deltaObject, ...subDelta}
+    return  delta.b * delta.b - 4 * delta.a * delta.c;
+  }
+}
+
+// delta = b*b - 4*a*c
+const getDelta = getDeltaFunction({ a: 1, b: 2, c: 3 });
+console.log(getDelta()); // -8
+console.log(getDelta({ b: 12 })); // 132
+console.log(getDelta({ a: 4, b: 10 })); // 52
+console.log(getDelta({ a: 4, b: 0 })); // -48
+
+// The nested function returns NaN if any of the arguments are missing
+console.log(getDeltaFunction({ a: 1, c: 3 })()); // NaN
+console.log(getDeltaFunction({ a: 1, c: 3 })({ b: 15 })); // 213
+
+// 2
+function positiveSum(numbers) {
+  let numbersSum = 0;
+
+  numbers.forEach(function(number) {
+    if (number > 0) {
+      numbersSum = numbersSum + number;
+    }
+  })
+
+  return numbersSum;
+}
+
+console.log(positiveSum([1, 5, 10]));
+console.log(positiveSum([-1, 5, -10]));
+
+// 3
+function countSheep(shed) {
+  let presentSheep = 0;
+
+  shed.filter(function(sheep) {
+    if (sheep === true) {
+      presentSheep = presentSheep + 1;
+    }
+  })
+
+  return presentSheep;
+}
+
+const shed = [true, false, true, false, true];
+console.log(countSheep(shed)); //3
+
+// 4
+function findNeedle(haystack) {
+  return 'found the needle at position ' + haystack.findIndex((word) => word === 'needle');
+}
+
+const haystack = ["hay", "junk", "hay", "hay", "moreJunk", "needle", "randomJunk"];
+console.log(findNeedle(haystack));
+
+// 5
+function doubleNumbers(numbers){
+  return numbers.map(function(number) {
+    return number * 2;
+  })
+}
+
+console.log(doubleNumbers([1, 2, 3]));
+
+// 6
+function invertNumbers(numbers) {
+  return numbers.map(function(number) {
+    return -number;
+  })
+}
+
+console.log(invertNumbers([1, -2, 3, -4, 5]));
+
+// 7
+function getAverage(marks){
+  let averageMarks = 0;
+
+  marks.forEach(function(mark) {
+    averageMarks += mark;
+  })
+
+  return Math.floor(averageMarks / marks.length);
+}
+
+console.log(getAverage([3, 4, 5]));
+
+// 8
+function isStringInArray(array) {
+  return !!array.find(function(edible) {
+    return typeof edible === 'string';
+  })
+}
+
+console.log(isStringInArray([4, 'Orange'], 'Apple')); // true
+console.log(isStringInArray([3, 4], 'Potato')); // false
+
+// 9
+function forEach(array, functionName) {
+  for (let i = 0; i < array.length; ++i) {
+    functionName(array[i], i.toString());
+  }
+}
+
+const vegetables = ['Carrot', 'Cabbage', 'Onion'];
+
+function printVegetable(vegetable, index) {
+  console.log(vegetable, index);
+}
+
+forEach(vegetables, printVegetable);
+// Carrot 0
+// Cabbage 1
+// Onion 2
+
+const fruits = ['Apple', 'Orange', 'Watermelon'];
+forEach(fruits,
+    function(fruit, index) {
+      console.log(fruit, index);
+    }
+);
+// Apple 0
+// Orange 1
+// Watermelon 2
+
+// 10
+function executeAfterFiveSeconds(functionName) {
```

Comment:
This argument is not a string, but `functionName` suggests it is. How do we call a function passed to a function that's supposed to be called?

Reply:
callback. Do I understand it correctly that I need to change a function name to 'callback' in this case?